codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/protocol/aws/AwsJson10ProtocolGenerator.java

```diff
@@ -86,10 +97,156 @@ private void generateSharedDeserializers(GenerationContext context, GoWriter wri
                     op.getOutputShape()));
             shared.addAll(shapes);
         }
+        var errorShapes = generateErrorDeserializers(context, writer, ops);
+        shared.addAll(errorShapes);
+
         var generator = new JsonDeserializerGenerator(context.getModel(), context.getSymbolProvider());
         writer.write(generator.generate(shared));
+
+        generateOperationErrorDeserializers(context, writer, ops);
+
+        writer.write(getProtocolErrorInfo());
+        writer.write(GET_AWS_QUERY_ERROR_CODE);
+    }
+
+    private Set<Shape> generateErrorDeserializers(GenerationContext context, GoWriter writer, Set<OperationShape> ops) {
+        Set<Shape> errorShapes = new HashSet<>();
```

Comment:
Yeah, I found that to be an issue with a lot of the stream operations that are amongst a lot of the different classes, and so I chose to go with the simple format since it's easier to read and understand. If you think the stream might be better and I should replace the code, let me know, but yeah I think it only saves some space in theory.